extractor/src/main.py

```diff
@@ -187,26 +187,23 @@ def broadcast_events(dataframe, analysis, batch_id=None):
                 event["resource_type"] = resource_type
                 event["resource_id"] = resource_id
                 event["record"] = record
-                try:
-                    producer.produce_event(
-                        topic=f"{PRODUCED_TOPIC_PREFIX}{batch_id}",
-                        event=event
-                    )
-                except KafkaException as e:
-                    if e.args[0].code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
-                        logger.debug(
-                            f"batch {batch_id} may have been deleted: extract topic does not exist"
-                        )
-                        break
-                    else:
-                        logger.error(format_traceback())
-                        break
+                producer.produce_event(
+                    topic=f"{PRODUCED_TOPIC_PREFIX}{batch_id}",
+                    event=event
+                )
                 count += 1
         except EmptyResult as e:
             logger.warn(
                 e,
                 extra={"resource_id": resource_id, "batch_id": batch_id}
             )
+        except KafkaException as e:
+            if e.args[0].code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
+                logger.warn(
+                    f"batch {batch_id} may have been deleted: extract topic does not exist"
+                )
```

Comment:
same comment as before: don't you think we should return before setting the batch counter in redis ? since the batch has been canceled, the counter should not be initialized at all

Reply:
Sorry I thought I answered but since I don't delete the counter, I think it would be nice to know what has been extracted exactly before the batch stopped.

Reply:
ok sounds good!